metrics/prometheus.go

```diff
@@ -210,6 +223,10 @@ func (p prometheusRec) IncCircuitbreakerState(state string) {
 	p.cbStateChanges.WithLabelValues(p.id, state).Inc()
 }
 
+func (p prometheusRec) SetCircuitbreakerCurrentCondition(condition int) {
```

Comment:
The problem with this metric approach is that the user needs to be aware of what is the number when it queries, this is not very natural and could lead to easy misinterpretation of the metrics.

To have the state of a resource in a metric what I have seen as a standardized way is that the state is in a label and when one of the states is set to "active" is set to `1` and the others to `0`. Is the same approach as the state counter but with a gauge of `(0,1)`.

What do you think?

Reply:
Yes, I think that your proposal to follow the conventions you have seen elsewhere is worth adopting. I'll make the necessary refactoring to implement this design.